File: algorithm_practice_typescript/1431.ts
function kidsWithCandies(candies: number[], extraCandies: number): boolean[] {
    const highest = Math.max(...candies);
    return candies.map(candy => candy +extraCandies >= highest)
}

const candies = [2,3,5,1,3]
const extraCandies = 3

console.log(kidsWithCandies(candies, extraCandies));

File: algorithm_practice_python/leetcode/arrays101/1431.py
from typing import List


class Solution:
    def kidsWithCandies(self, candies: List[int], extraCandies: int) -> List[bool]:
        ans = []
        max_amount = max(candies)
        for i in candies:
            if i + extraCandies >= max_amount:
                ans.append(True)
            else:
                ans.append(False)
        return ans

    def sol2(self, candies, extra):
        max_candies = max(candies)
        return [True if i+extra >= max_candies else False for i in candies]

if __name__ == '__main__':
    candies = [2, 3, 5, 1, 3]
    extraCandies = 3

    ans = Solution().sol2(candies, extraCandies)
    print(f'{ans}, [true,true,true,false,true]')
